### PR TITLE
[FLINK-30263] Introduce schemas table in table store

### PR DESCRIPTION
**Pull request by zjureel**
Introduce schemas table in table store, user can query historical schemas of table through sql: SELECT * FROM Table$schemas

### CODE REVIEW

File: docs/content/docs/development/query-table.md [hunk 1]
```diff
@@ -110,6 +110,32 @@ SELECT * FROM MyTable$snapshots;
...
+
+You can query the historical schemas of the table through Flink SQL.
+
+```sql
+SELECT * FROM MyTable$schemas;
+
++-----------+----------------------------------------+----------------+--------------+-----------+-----------+
+| schema_id |                  fields                | partition_keys | primary_keys |  options  | commonts  |
++-----------+----------------------------------------+----------------+--------------+-----------+-----------+
+|   1       | [{"id":0,"name":"a","type":"BIGINT"... |     ["k1"]     |  ["k1", "k2] |    {}     | my table  |
```
Comment (JingsongLi) [inline docs/content/docs/development/query-table.md:123]: In SQL-CLI, do we have option to show long field? I think we can use it here.
`sql-client.display.max-column-width`

File: flink-table-store-core/src/main/java/org/apache/flink/table/store/table/metadata/SchemasTable.java [hunk 2]
```diff
@@ -0,0 +1,258 @@
...
+            if (projection != null) {
+                rows =
+                        Iterators.transform(
+                                rows, row -> ProjectedRowData.from(projection).replaceRow(row));
+            }
+            return new IteratorRecordReader<>(rows);
+        }
+
+        private RowData toRow(TableSchema schema) {
+            List<GenericMapData> fields = new ArrayList<>(schema.fields().size());
```
Comment (JingsongLi) [inline flink-table-store-core/src/main/java/org/apache/flink/table/store/table/metadata/SchemasTable.java:210]: Useless `fields`

File: flink-table-store-core/src/main/java/org/apache/flink/table/store/table/metadata/SchemasTable.java [hunk 3]
```diff
@@ -0,0 +1,258 @@
...
+                                Map<StringData, StringData> field = new LinkedHashMap<>(4);
+                                field.put(
+                                        StringData.fromString("id"),
+                                        StringData.fromString(String.valueOf(f.id())));
+                                field.put(
+                                        StringData.fromString("name"),
+                                        StringData.fromString(f.name()));
+                                field.put(
+                                        StringData.fromString("type"),
+                                        StringData.fromString(f.type().logicalType().toString()));
```
Comment (JingsongLi) [inline flink-table-store-core/src/main/java/org/apache/flink/table/store/table/metadata/SchemasTable.java:223]: We can use `DataFieldSerializer.typeToJson`. Maybe we can implement `DataType.toString`.

**Review by JingsongLi — COMMENTED** [review]
Thanks @zjureel for the contribution, left comments.

File: flink-table-store-core/src/main/java/org/apache/flink/table/store/table/metadata/SchemasTable.java (hunk 3 above)
Comment (zjureel) [reply flink-table-store-core/src/main/java/org/apache/flink/table/store/table/metadata/SchemasTable.java]: Done

File: flink-table-store-core/src/main/java/org/apache/flink/table/store/table/metadata/SchemasTable.java (hunk 2 above)
Comment (zjureel) [reply flink-table-store-core/src/main/java/org/apache/flink/table/store/table/metadata/SchemasTable.java]: Done

File: docs/content/docs/development/query-table.md (hunk 1 above)
Comment (zjureel) [reply docs/content/docs/development/query-table.md]: > In SQL-CLI, do we have option to show long field? I think we can use it here. `sql-client.display.max-column-width`

The config `sql-client.display.max-column-width` is supported in `streaming` only, I have created a issue https://issues.apache.org/jira/browse/FLINK-30303 in `Flink`

**Pull request merged**